flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer08.java

```diff
@@ -295,7 +296,7 @@ public void open(Configuration parameters) throws Exception {
 
 	@Override
 	public void run(SourceContext<T> sourceContext) throws Exception {
-		if (fetcher != null) {
+		if (fetcher != null && isAutoCommitEnabled()) {
```

Comment:
This is not correct!
If the fetcher != null and auto commit disabled, `fetcher.run()` (line 316) is never called! This means that the fetcher is doing anything at all!

I'm not sure how the integration tests were passing with that change! (maybe there is no test testing auto commit enable = false.)